waitlist/__main__.py

```diff
@@ -0,0 +1,133 @@
+        index = attributes.index(attribute)
+
+        duplicates_free.append(_data[index])
+
+    return duplicates_free
+
+
+def applyCOMMrules(nr_waitlist: list, stake_waitlist: list, chunk_sizes: tuple):
+    nr_chunk_size, stake_chunk_size = chunk_sizes
+    nr_index, stake_index = 0, 0
+
+    ordered_waitlist = []
+
+    nr_waitlist.sort(key=lambda x: x.date)
+    stake_waitlist.sort(key=lambda x: x.date)
+
+    while len(ordered_waitlist) != (len(nr_waitlist) + len(stake_waitlist)):
+        if nr_index < len(nr_waitlist):
+            ordered_waitlist.extend(nr_waitlist[nr_index : nr_index + nr_chunk_size])
+            nr_index += nr_chunk_size
+
+        if stake_index < len(stake_waitlist):
+            ordered_waitlist.extend(
+                stake_waitlist[stake_index : stake_index + stake_chunk_size]
+            )
+            stake_index += stake_chunk_size
+
+    return ordered_waitlist
+
+
+@click.command()
+@click.option(
+    "--nrfile",
+    default="network_register.xlsx",
+    help="Network register file (.xlsx)",
+)
+@click.option(
+    "--waitlist",
+    default="network_waitlist.xlsx",
+    help="Network waitlist file (.xlsx)",
+)
+@click.option(
+    "--output-file",
+    default="final_waitlist.xlsx",
+    help="Output file (.xlsx)",
+)
+def main(nrfile: str, waitlist: str, output_file: str):
+    # Loading onboarding waitlist (from Dune)
+    onboarding_data = dune.run_query_dataframe(QueryBase(env.get("DUNE_QUERY_ID")))
+    onboardings = DuneEntry.fromDataFrame(onboarding_data)
+    unique_onboarding = remove_duplicates(onboardings, "safe_address", True)
+    addresses_from_onboarding = [e.safe_address for e in unique_onboarding]
+
+    # Loading registration data (from Andrius)
+    registrations = RegistrationEntry.fromXLSX(nrfile)
+    unique_registrations = remove_duplicates(registrations, "safe_address", True)
+
+    # Loading network waitlist (from Cryptpad)
+    network_waitlist = NetworkWaitlistEntry.fromXLSX(waitlist)
+    eligible_addresses = [e.safe_address for e in network_waitlist if e.eligible]
+
+    print(f"Eligible addresses\t\t{len(eligible_addresses)}")
+
+    # Cleanup registrations to get only valid candidates
+    waitlist_candidates = [
+        e for e in unique_registrations if e.safe_address not in eligible_addresses
+    ]
+    print(f"Candidates after cleanup\t{len(waitlist_candidates)}")
+
+    # Filtering candidates by stake and NFT ownership
+    waitlist = []
+    for c in waitlist_candidates:
+        if c.safe_address not in addresses_from_onboarding:
+            continue
+
+        index = addresses_from_onboarding.index(c.safe_address)
+        candidate = unique_onboarding[index]
+        candidate.node_address = c.node_address
+
+        if candidate.wxHOPR_balance < 10000:
+            continue
+
+        if candidate.wxHOPR_balance < 30000 and not candidate.nr_nft:
+            continue
+
+        if not candidate.node_address.startswith("0x"):
```

Comment:
This does not mean that the address is valid, it's just more likely to be valid. 🍻

Reply:
Is there a way to check that the address is actually valid ?